common/java_deps/java_deps.py

```diff
@@ -40,4 +45,5 @@ def tarfile_remove_mtime(info):
 
 with tarfile.open(distribution_tgz_location, 'w:gz', dereference=True) as tgz:
     for fn, arcfn in sorted(moves.items()):
-        tgz.add(fn, arcfn.replace('{pom_version}', version), filter=tarfile_remove_mtime)
+        path = "\\\\?\\" + os.path.abspath(fn) if platform.system() == WINDOWS else fn
```

Comment:
Maven/Coursier can generate extremely long filenames.

Prepending `\\?\` to the path marks it as an 'extended path' on Windows, allowing it to bypass Windows' 260 character limit.